### PR TITLE
Fix BelongsToSelect key with custom OptionLabel

### PR DESCRIPTION
**Pull request by MissaelAnda**
When using a custom OptionLabelFromRecord it used the collection id instead of model's id, fixed with mapWithKeys.

### CODE REVIEW

**Comment by pxlrbt** [comment]
This needs to be fixed in `BelongsToManySelect` and `BelongsToManyCheckboxes` too.
See this commit:
https://github.com/laravel-filament/filament/pull/1658/commits/138793fe20e94e1d78bd9908b83d8e01fcdf6917

File: packages/forms/src/Components/BelongsToSelect.php [hunk 1]
```diff
@@ -126,7 +128,9 @@ public function relationship(string | Closure $relationshipName, string | Closur
             if ($component->hasOptionLabelFromRecordUsingCallback()) {
                 return $relationshipQuery
                     ->get()
-                    ->map(fn (Model $record) => $component->getOptionLabelFromRecord($record))
+                    ->mapWithKeys(fn (Model $record) => [
+                        $record->{$relationship->getOwnerKeyName()} => $component->getOptionLabelFromRecord($record)
```
Comment (pxlrbt) [inline packages/forms/src/Components/BelongsToSelect.php:132]: Do we need `$relationship->getOwnerKeyName()` here? Isn't that the same as `$record->getKey()`?
Before we had:

```suggestion
                        $record->getKey() => $component->getOptionLabelFromRecord($record)
```

File: packages/forms/src/Components/BelongsToSelect.php [hunk 2]
```diff
@@ -99,7 +99,9 @@ public function relationship(string | Closure $relationshipName, string | Closur
...
```
Comment (pxlrbt) [inline packages/forms/src/Components/BelongsToSelect.php:103]:
```suggestion
                        $record->getKey() => $component->getOptionLabelFromRecord($record)
```

**Comment by MissaelAnda** [comment]
Added the solution to all BelongsTo coponents and replaced $relationship->getOwnerKeyName() with getKey()

File: packages/forms/src/Components/BelongsToManyMultiSelect.php [hunk 3]
```diff
@@ -71,7 +71,9 @@ public function relationship(string | Closure $relationshipName, string | Closur
             if ($component->hasOptionLabelFromRecordUsingCallback()) {
                 return $relationshipQuery
                     ->get()
-                    ->map(fn (Model $record) => $component->getOptionLabelFromRecord($record))
+                    ->mapWithKeys(fn (Model $record) => [
+                        $record->{$relatedKeyName} => $component->getOptionLabelFromRecord($record)
```
Comment (pxlrbt) [inline packages/forms/src/Components/BelongsToManyMultiSelect.php:75]: Do we need `$relatedKeyName` here or can we use `->getKey()` like on the other components?

```suggestion
                        $record->getKey() => $component->getOptionLabelFromRecord($record)
```

**Comment by danharrin** [comment]
I actually think `getOwnerKeyName()` is a more correct approach than `getKey()`, as you may be using a non-primary column as the owner key in the relationship.

**Comment by pxlrbt** [comment]
> I actually think `getOwnerKeyName()` is a more correct approach than `getKey()`, as you may be using a non-primary column as the owner key in the relationship.

Will you? Isn't that just the key of the related `$record`

**Comment by MissaelAnda** [comment]
getOwnerKeyName() will return the primary key name, whereas getKey() will return the actual value of that column, I think it's the same result but with one less step (except if the model has a mutator on the primary key)

**Comment by pxlrbt** [comment]
I guess `->getKey()` won't work with relations that don't use the primary key of the relation, right? Is this even a thing? 😅 Maybe your first version was better.

**Comment by MissaelAnda** [comment]
Ohhh you're right, reverting now

**Comment by danharrin** [comment]
Thanks gents

**Pull request merged**